controllers/chat/ChatController.js

```diff
@@ -13,9 +13,133 @@ const BetterSocialConstantListUtils = require('../../services/bettersocial/const
 const {CHANNEL_TYPE} = require('../../helpers/constants');
 const ChatAnonUserInfoFunction = require('../../databases/functions/chatAnonUserInfo');
 const BetterSocialCore = require('../../services/bettersocial');
+const {Op} = require('sequelize');
+const crypto = require('crypto');
 
 const v = new Validator();
 
+const generate_channel_id_for_anon_chat = (owner, member) => {
+  const hash = crypto.createHash('sha256');
+  hash.update(owner, member);
+  return hash.digest('hex');
+};
+
+const is_all_anon_user = async (members) => {
+  // check all members users are anonymous
+  let users = await User.findAll({
+    where: {
+      user_id: {
+        [Op.in]: members
+      },
+      is_anonymous: false
+    }
+  });
+  return users.length === 0 ? true : false;
+};
+
+const generate_anon_user_info = () => {
+  const emoji = BetterSocialConstantListUtils.getRandomEmoji();
+  const color = BetterSocialConstantListUtils.getRandomColor();
+  return {
+    anon_user_info_color_code: color.code,
+    anon_user_info_color_name: color.color,
+    anon_user_info_emoji_code: emoji.emoji,
+    anon_user_info_emoji_name: emoji.name
+  };
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This function generates anonymous user info, including a random color and emoji from the `BetterSocialConstantListUtils` module. It returns an object with the color and emoji codes and names.
```javascript
const generate_anon_user_info = () => {
  const emoji = BetterSocialConstantListUtils.getRandomEmoji();
  const color = BetterSocialConstantListUtils.getRandomColor();
  return {
    anon_user_info_color_code: color.code,
    anon_user_info_color_name: color.color,
    anon_user_info_emoji_code: emoji.emoji,
    anon_user_info_emoji_name: emoji.name
  };
};
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->